File: backend/Turistando/Turistando.WebApi/Models/Roteiro.cs
using System.Collections.Generic;
using System.Runtime;

namespace Turistando.WebApi.Models
{
    public class Roteiro
    {
        public Roteiro()
        {
            this.ReservaRoteiros = new List<ReservaRoteiro>();
            this.RoteiroCategorias = new List<RoteiroCategoria>();
        }

        public int Id { get; set; }
        public int IdGuia { get; set; }
        public string Cidade { get; set; }
        public string Estado { get; set; }
        public string Descricao { get; set; }
        public double Valor { get; set; }
        public double Tempo { get; set; }

        public virtual Usuario Guia { get; set; }
        public virtual ICollection<ReservaRoteiro> ReservaRoteiros { get; set; }
        public virtual ICollection<RoteiroCategoria> RoteiroCategorias { get; set; }
    }
}

File: frontend/turistando-react/src/models/roteiro.ts
import {ReservaRoteiro} from "./reserva-roteiro";
import {RoteiroCategoria} from "./roteiro-categoria";
import {Usuario} from "./usuario";

export class Roteiro
{
    constructor(idGuia: number)
    {
        this.Id = 0;
        this.IdGuia = idGuia;
        this.Nome = '';
        this.Imagem = 'no-image.jpg';
        this.Cidade = '';
        this.Estado = '';
        this.Descricao = '';
        this.Valor = 0;
        this.Tempo = 0;
        this.ReservaRoteiros = new Array<ReservaRoteiro>();
        this.RoteiroCategorias = new Array<RoteiroCategoria>();
    }

    public Id: number;
    public IdGuia: number;
    public Nome: string;
    public Imagem: string;
    public Cidade: string;
    public Estado: string;
    public Descricao: string;
    public Valor: number;
    public Tempo: number;

    public Guia: Usuario;
    public ReservaRoteiros: ReservaRoteiro[];
    public RoteiroCategorias: RoteiroCategoria[];
}
